src/components/workspace/workspace.tsx

```diff
@@ -1,13 +1,124 @@
-import React from "react";
-import './workspace.css'
-
-const Workspace = ()=>{
-    return (
-        <div className="workspace">
-            <svg className="canvas" viewBox="0 0 1200 600" version="1.1" xmlns="http://www.w3.org/2000/svg">
-            </svg>
-        </div>
-    )
-}
-
-export default Workspace;
\ No newline at end of file
+import React, { useCallback, useRef, useState } from "react";
+import {
+  DropTargetMonitor,
+  useDragDropManager,
+  useDrop,
+  XYCoord,
+} from "react-dnd";
+import { DragTypeEnum } from "../../data/dragTypeEnum";
+import { DrawnShapeModel } from "../../data/drawnShapeModel";
+import ShapeModel from "../../data/shapeModel";
+import WorkspaceModel from "../../data/workspaceModel";
+import { mapPixelsToSvgCoordinate } from "../../helpers/ui/svg";
+import DrawnShape from "./drawnShape/drawnShape";
+import "./workspace.css";
+
+const Workspace = () => {
+  const [workspace, setWorkspace] = useState(new WorkspaceModel());
+  const dragDropManager = useDragDropManager();
+  const monitor = dragDropManager.getMonitor();
+  const svgViewportWidth = 1200;
+  const svgViewportHeight = 600;
+  const stateRef = useRef<WorkspaceModel>();
+  stateRef.current = workspace;
+
+  monitor.subscribeToOffsetChange(() => {
```

Comment:
The `monitor` related logic, probably, could be extracted to a custom hook, to make it testable and to simplify the `Workspace` component.